python/ml4ir/base/pipeline.py

```diff
@@ -265,9 +324,122 @@ def get_relevance_model(self, feature_layer_keys_to_fns={}) -> RelevanceModel:
         """
         raise NotImplementedError
 
+    def create_pipeline_for_kfold(self, args):
+        raise NotImplementedError
+
+    def kfold_analysis(self, base_logs_dir, run_id, num_folds, pvalue_threshold=0.1, metrics=None):
+        """
+        Aggregate results of the k-fold runs and perform t-test on the results between old(prod model) and
+        new model's w.r.t the specified metrics.
+
+        Parameters
+        ----------
+        base_logs_dir : int
+            Total number of folds
+        run_id : int
+            current fold number
+        num_folds: int
+            Total number of folds
+        metrics: set()
+            Set of metrics to include in the kfold analysis
+        """
+        if not metrics:
+            self.logger.warning("No metrics specified for Kfold CV analysis")
+            return
+        rows = []
+        for i in range(num_folds):
+            row = {"fold": i}
+            logs_dir = pathlib.Path(base_logs_dir) / run_id / "fold_{}".format(i) / run_id
+            with open(logs_dir / "_SUCCESS", 'r') as f:
+                for line in f:
+                    parts = line.split(',')
+                    if parts[0] in metrics:
+                        row[parts[0]] = float(parts[1])
+            rows.append(row)
+        results = pd.DataFrame(rows)
+        self.logger.info(results)
+        # calculate statistical paired t-test
+        for dataset in ['train', 'val', 'test']:
+            t_test_stat, pvalue = stats.ttest_rel(results["{}_old_MRR".format(dataset)],
+                                                  results["{}_new_MRR".format(dataset)])
+            self.logger.info("{}_t_test_stat={}".format(dataset, t_test_stat))
+            self.logger.info(
+                "{}_pvalue={} --> (statistically significant={})".format(dataset, pvalue, pvalue < pvalue_threshold))
```

Comment:
@mohazahran You need to also find a way to add this to `metrics_dict`.

Reply:
The reason I keep pushing for this is because you are not thinking in terms of training models at scale. How would you fetch these values from the logs using, say a script?